WalletWasabi/WabiSabi/Client/CoinJoinClient.cs

```diff
@@ -447,9 +456,31 @@ internal static ImmutableList<SmartCoin> SelectCoinsForRound(IEnumerable<SmartCo
 		var privateCoins = filteredCoins
 			.Where(x => x.HdPubKey.AnonymitySet >= anonScoreTarget)
 			.ToArray();
-		var nonPrivateCoins = filteredCoins
-			.Where(x => x.HdPubKey.AnonymitySet < anonScoreTarget)
-			.ToArray();
+
+		// If we want to isolate red coins from each other, then only let a single red coin get into our selection candidates.
+		SmartCoin[]? nonPrivateCoins = null;
+		if (redCoinIsolation)
+		{
+			var nonPrivateCandidates = filteredCoins
+					.Where(x => x.HdPubKey.AnonymitySet < anonScoreTarget && x.HdPubKey.AnonymitySet != 1)
+					.ToList();
+
+			var randomRed = filteredCoins
+				.Where(x => x.HdPubKey.AnonymitySet == anonScoreTarget)
```

Comment:
I guess here you mean 
```suggestion
				.Where(x => x.HdPubKey.AnonymitySet == 1)
```

Reply:
I guess so, too.

Or 
`.Where(x => x.HdPubKey.AnonymitySet < anonScoreTarget)`

Depends on what we want to call a `Red Coin`.

Reply:
> Or `.Where(x => x.HdPubKey.AnonymitySet < anonScoreTarget)`
> 
> Depends on what we want to call a `Red Coin`.

No a red coin is a coin that has 1 anonset